uaclient/snap.py

```diff
@@ -93,3 +95,44 @@ def get_config_option_value(key: str) -> Optional[str]:
         return out.strip()
     except exceptions.ProcessExecutionError:
         return None
+
+
+def get_snap_package_info_tracking(package: str) -> Optional[str]:
+    out, _ = system.subp(["snap", "info", package])
+    match = re.search(r"tracking:\s+(?P<tracking>.*)", out)
+    if match:
+        return match.group("tracking")
+    return None
+
+
+class SnapPackage:
+    def __init__(self, name, version, rev, tracking, publisher, notes):
+        self.name = name
+        self.version = version
+        self.rev = rev
+        self.tracking = tracking
+        self.publisher = publisher
+        self.notes = notes
+
+
+def get_installed_packages() -> List[SnapPackage]:
+    out, _ = system.subp(["snap", "list"])
+    apps = out.splitlines()
+    apps = apps[1:]
+    snaps = []
+    for line in apps:
+        pkg = line.split()
+        snap = SnapPackage(
+            name=pkg[0],
+            version=pkg[1],
+            rev=pkg[2],
+            tracking=pkg[3],
+            publisher=pkg[4],
```

Comment:
We might have some special character for `publisher`, like double `**`. Should we remove those ?

Reply:
I do not see any harm, considering they indicate a verified publisher